### PR TITLE
Add initial GradientBoostingClassifier support (binary classification)

### PR DESCRIPTION
**Pull request by smokedlab**
Add initial GradientBoostingClassifier support (binary classification only)

Scope:
- adds emlearn/gbtrees.py as a separate module
- supports inline inference mode
- supports predict_proba (sigmoid link for binary classification)
- adds tests comparing predictions against scikit-learn

Not included in this PR:
- loadable inference mode
- multi-class classification (softmax)
- GradientBoostingRegressor
- HistGradientBoostingClassifier

### CODE REVIEW

**Comment by smokedlab** [comment]
Hi @jonnor,

just wanted to check whether you had a chance to look at this PR.

I'm interested in continuing work on Gradient Boosted Trees in emlearn and would appreciate any feedback on whether this direction fits the project.

Thanks!

**Comment by jonnor** [comment]
Hi @smokedlab and thank you for this MR. The changes look to be overall in the right direction, both in terms of structure and the testing.

- Could you extend this to handle more than 2 classes? That would make it useful in the general case.
- Is there a way to reuse a bit more code from trees.py? In particular I am thinking about flatten_trees - that seems very similar. And if it is not feasible / leads to more complexity, then it would be good to highlight the reason with a code comment.
